src/main/java/com/ifactory/press/db/solr/processor/UpdateDocValuesProcessor.java

```diff
@@ -21,166 +21,184 @@
+                        }
+
+                        doc.addField(valueField, lvalue);
+                        // LOG.debug("retrieved doc value %d", lvalue);
+                    } else {
+                        doc.addField(valueField, 0);
+                        // LOG.debug("Initializing doc value to 0");
+                    }
+                } else {
+                    // LOG.debug(String.format("%s found in input document: %s", valueField, doc.get(valueField)));
+                }
+            }
         } else {
-          // LOG.debug(String.format("%s found in input document: %s", valueField, doc.get(valueField)));
+            for (String valueField : valueFields) {
+                // new document, inserted without dv values: default to zero
+                doc.addField(valueField, 0);
+            }
+        }
+        List<SolrInputDocument> childDocuments = doc.getChildDocuments();
+        if (childDocuments != null) {
+            for (SolrInputDocument childDoc : childDocuments) {
+                retrieveDocValuesHelper(valueFields, childDoc, searcher);
+            }
         }
-      } 
-    } else {
-      for (String valueField : valueFields) {
-        // new document, inserted without dv values: default to zero
-        doc.addField(valueField, 0);
-      }
-    }
-    List<SolrInputDocument> childDocuments = doc.getChildDocuments();
-    if (childDocuments != null) {
-      for (SolrInputDocument childDoc : childDocuments) {
-        retrieveDocValuesHelper(valueFields, childDoc, searcher);
-      }
-    }
-  }
-  
-  private String getStringValue (SolrInputDocument solrInputDocument, String keyField) {
-    Object o = solrInputDocument.getFieldValue(keyField);
-    if (o == null) {
-      return null;
-    }
-    if (! (o instanceof String)) {
-      throw new SolrException(ErrorCode.BAD_REQUEST, "Value of DocValue key field " + keyField + " must be a string");
     }
-    return (String) o;
-  }
-  
-  private long getLongValue(SolrInputDocument solrInputDocument, String valueField) {
-    Object o = solrInputDocument.getFieldValue(valueField);
-    if (o == null) {
-      throw new SolrException(ErrorCode.BAD_REQUEST, "no value for DocValues value field " + valueField);
+
+    private String getStringValue(SolrInputDocument solrInputDocument, String keyField) {
+        Object o = solrInputDocument.getFieldValue(keyField);
+        if (o == null) {
+            return null;
+        }
+        if (!(o instanceof String)) {
+            throw new SolrException(ErrorCode.BAD_REQUEST, "Value of DocValue key field " + keyField + " must be a string");
+        }
+        return (String) o;
     }
-    try {
-      return Long.parseLong(o.toString());
-    } catch (NumberFormatException e) {
-      throw new SolrException(ErrorCode.BAD_REQUEST, "Value of DocValue valuefield " + valueField + " must be an integer, not " + o);
+
+    private long getLongValue(SolrInputDocument solrInputDocument, String valueField) {
```

Comment:
Interesting historical note: Right now this processor (UpdateDocValues) only handles integer fields which is very lame. At one point I tried to get it to support floats but then got pulled away on something else. Would be interesting to see in the future if we will need to extend this function.